src/main/java/org/openhab/binding/zigbee/converter/ZigBeeConverterColorColor.java

```diff
@@ -88,45 +146,99 @@ public void initializeConverter() {
     public void disposeConverter() {
         clusterColorControl.removeAttributeListener(this);
         clusterLevelControl.removeAttributeListener(this);
+        clusterOnOff.removeAttributeListener(this);
     }
 
     @Override
     public void handleRefresh() {
-        clusterColorControl.getCurrentHue(0);
-        clusterColorControl.getCurrentSaturation(0);
+        if(supportsHue) {
+            clusterColorControl.getCurrentHue(0);
+            clusterColorControl.getCurrentSaturation(0);
+        }
+        else {
+            clusterColorControl.getCurrentX(0);
+            clusterColorControl.getCurrentY(0);
+        }
+
         clusterLevelControl.getCurrentLevel(0);
+        clusterOnOff.getOnOff(0);
     }
 
     @Override
     public Runnable handleCommand(final Command command) {
         return new Runnable() {
+            protected void changeOnOff(OnOffType onoff) throws InterruptedException, ExecutionException {
```

Comment:
Why not make all these methods in the main class rather than this runnable? I plan to change the way this works (I've actually already done so) so I would prefer them as just private methods in the converter class (why are they just protected by the way?).

Reply:
No particular reason. These are called only from the Runnable, so I had put them there, but I will move them to the main class. Will make them private also.